client/src/api/block_builder/input_selection/new/requirement/base_token.rs

```diff
@@ -0,0 +1,160 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::HashSet;
+
+use super::{InputSelection, OutputInfo, Requirement};
+use crate::{
+    block::output::{
+        unlock_condition::UnlockCondition, AliasOutputBuilder, FoundryOutputBuilder, NftOutputBuilder, Output, Rent,
+    },
+    error::{Error, Result},
+    secret::types::InputSigningData,
+};
+
+// TODO checked operations ?
+
+pub(crate) fn base_token_sums(selected_inputs: &[InputSigningData], outputs: &[OutputInfo]) -> (u64, u64) {
+    let inputs_sum = selected_inputs.iter().map(|input| input.output.amount()).sum::<u64>();
+    let outputs_sum = outputs.iter().map(|output| output.output.amount()).sum::<u64>();
+
+    (inputs_sum, outputs_sum)
+}
+
+impl InputSelection {
+    pub(crate) fn fulfill_base_token_requirement(&mut self) -> Result<(Vec<InputSigningData>, Option<Requirement>)> {
+        let (mut inputs_sum, mut outputs_sum) = base_token_sums(&self.selected_inputs, &self.outputs);
+        let mut newly_selected_inputs = Vec::new();
+        let mut newly_selected_ids = HashSet::new();
+
+        if inputs_sum >= outputs_sum {
+            return Ok((newly_selected_inputs, None));
+        }
+
+        // TODO don't pick burned things
+
+        // TODO if consolidate strategy: sum all the lowest amount until diff is covered.
+        // TODO this would be lowest amount of input strategy.
+        self.available_inputs
+            .sort_by(|left, right| left.output.amount().cmp(&right.output.amount()));
+
+        'overall: {
+            // 1. Basic with ED25519 address and nothing else
+            {
+                let inputs = self.available_inputs.iter().filter(|input| {
+                    if let Output::Basic(output) = &input.output {
+                        if let [UnlockCondition::Address(address)] = output.unlock_conditions().as_ref() {
```

Comment:
I would say with an expired timelock and or expiration where our address is in the expiration unlock condition it should be used the same way

Is that what was meant with `// 2. Basic with ED25519 address and other things ????`?  Could be merged I think

Reply:
We said we would filter for timelock and expiration at the beginning so we don't have to do it everywhere so yeah I guess you're right, it should be fine to do it all together